lib/travis/logs/services/fetch_log.rb

```diff
@@ -0,0 +1,41 @@
+module Travis
+  module Logs
+    module Services
+      class FetchLog
+        def initialize(database: nil)
+          @database = database || Travis::Logs.database_connection
+        end
+
+        attr_reader :database
+        private :database
+
+        def run(job_id: nil, id: nil)
+          return nil if job_id.nil? && id.nil?
+          if job_id && id
+            raise ArgumentError, 'only one of job_id or id allowed'
+          end
+
+          result = nil
```

Comment:
I was mostly asking since the conditionals above guarantee that one or the other of the two following lines will be called, so there's no case where this nil will be used. If you think it makes it clearer, I think it's fine to keep it, though.